File: src/com/designpatterns/creational/factory/Square.java
package com.designpatterns.creational.factory;

public class Square implements IShape {

    public static final String TAG = "Square";

    float side;

    public Square(float side) {
        this.side = side;
    }

    @Override
    public float area() {
        return side * side;
    }

    @Override
    public float perimeter() {
        return 4 * side;
    }

    @Override
    public void draw() {
        System.out.printf("Square:: area:%f, perimeter:%f%n", area(), perimeter());
    }
}

File: src/com/designpatterns/creational/factory/kotlin/Square.kt
package com.designpatterns.creational.factory.kotlin

class Square(var side: Float) : IShape {

    override fun area(): Float {
        return side * side
    }

    override fun perimeter(): Float {
        return 4 * side
    }

    override fun draw() {
        System.out.printf("Square:: area:%f, perimeter:%f%n", area(), perimeter())
    }

    companion object {
        const val TAG = "Square"
    }
}